charon/tests/ui/matches.out

```diff
@@ -13,14 +13,19 @@ fn test_crate::test1(@1: test_crate::E1) -> bool
 
     @fake_read(x@1)
     match x@1 {
-        0 => {
-            @0 := const (true)
-        },
-        1 => {
+        0 | 1 => {
             @0 := const (true)
         },
-        2 => {
-            @0 := const (false)
+        _ => {
+            match x@1 {
+                2 => {
+                    @0 := const (false)
+                },
+                _ => {
+                    @fake_read(x@1)
+                    undefined_behavior
+                }
+            }
```

Comment:
Interesting, not sure what changed here. I did make _some_ changes to the MIR-lowering of or-patterns but they shouldn't have affected a simple case like this.

Reply:
This was caused by a bug in MIR generation in rustc. The fix is here: https://github.com/rust-lang/rust/pull/127028.